buildstockbatch/gcp/gcp.py

```diff
@@ -1192,8 +1199,15 @@ def main():
         elif args.crawl:
             batch.process_results(skip_combine=True, use_dask_cluster=False)
         else:
-            # TODO: check whether this job ID already exists. If so, don't try to start a new job, and maybe reattach
-            # to the existing one if it's still running.
+            existing_job = batch.get_existing_batch_job()
+            if existing_job:
+                job_status = existing_job.status.state.name
+                logger.error(
+                    f"A job with this ID ({batch.job_identifier}) already exists (status: {job_status}). "
+                    "Choose a new job_identifier or run with --clean to delete the existing job."
+                )
+                return
```

Comment:
We should check for the post-processing job, too, so that the user finds our right away rather than when the batch is done at some arbitrary point in the future.

Then, I think maybe these checks should be extracted out of `main()` (e.g., `batch.check_for_existing_jobs`).